### PR TITLE
Top level bug <bug

### PR DESCRIPTION
**Pull request by arjovanramshorst**

### CODE REVIEW

**Comment by wouterz** [comment]
it looks good but when i actually run the game and try it, i can still leave the top of the level. <feature, bug

File: core/src/test/java/sem/group15/bubblebobble/core/objects/FilledBubbleTest.java [hunk 1]
```diff
@@ -73,7 +79,7 @@ public void testHandleCollisionPlayerSlow(){
      */
     @Test
     public void testDraw() {
-        SpriteBatch batch = Mockito.mock(SpriteBatch.class);
+        SpriteBatch batch = mock(SpriteBatch.class);
         Texture texture = null;
```
Comment (DaanVermunt) [inline core/src/test/java/sem/group15/bubblebobble/core/objects/FilledBubbleTest.java]: why did you change Mockito.mock to mock everywhere, just for layout cleanness? <layout>

File: core/src/main/java/sem/group15/bubblebobble/core/objects/Gravity.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void update(float elapsed) {
     public void handleCollision(GameObject other) {
         if (other instanceof Floor) {
             float MAX_DIFF_LANDING = 10f;
-            if (between(overlapBottom(other), 0, MAX_DIFF_LANDING) && speedY < 0) {
+            if (getTop() < Gdx.graphics.getHeight() && between(overlapBottom(other), 0, MAX_DIFF_LANDING) && speedY < 0) {
                 setBottom(other.getTop());
```
Comment (DaanVermunt) [inline core/src/main/java/sem/group15/bubblebobble/core/objects/Gravity.java]: watch out for to long lines, it is now set to 120, maybe put an and statement on a new line <layout>

File: core/src/main/java/sem/group15/bubblebobble/core/objects/GameObject.java [hunk 3]
```diff
@@ -247,7 +247,11 @@ public final boolean between(float value, float low, float high) {
     public final void addNewObjectsTo(List<GameObject> gameObjects) {
         if (!newObjects.isEmpty()) {
             for (GameObject object : newObjects) {
-                gameObjects.add(object);
+                if(object instanceof Enemy) {
+                    gameObjects.add(1, object);
+                } else {
+                    gameObjects.add(object);
+                }
             }
```
Comment (DaanVermunt) [inline core/src/main/java/sem/group15/bubblebobble/core/objects/GameObject.java]: kind of a hackey way but seems to work fine

**Comment by DaanVermunt** [comment]
Over all looks good, I say go with it

File: core/src/main/java/sem/group15/bubblebobble/core/objects/GameObject.java (hunk 3 above)
Comment (MatthijsRijlaarsdam) [inline core/src/main/java/sem/group15/bubblebobble/core/objects/GameObject.java]: Totally. #embraceThatHackerWay

**Comment by MatthijsRijlaarsdam** [comment]
Please resolve the conflicts first, then we can check again <bug

File: core/src/test/java/sem/group15/bubblebobble/core/objects/FilledBubbleTest.java (hunk 1 above)
Comment (arjovanramshorst) [inline core/src/test/java/sem/group15/bubblebobble/core/objects/FilledBubbleTest.java]: Did my editor after importing Mockito.mock (And looks cleaner indeed)

**Comment by arjovanramshorst** [comment]
Fixed conflicts, should be ready for merging I guess
